src/em/prov/easyconnect/ec_util.cpp

```diff
@@ -24,25 +24,35 @@ void ec_util::init_frame(ec_frame_t *frame)
     frame->crypto_suite = 0x01; // Section 3.3 (Currently only 0x01 is defined)
 }
 
-ec_attribute_t *ec_util::get_attrib(uint8_t *buff, size_t len, ec_attrib_id_t id)
+const std::optional<const ec_attribute_t> ec_util::get_attrib(uint8_t *buff, size_t len, ec_attrib_id_t id)
```

Comment:
So I think you were right and we don't need the `const` for the optional return type. Removing it will also remove the need for the `const auto` as well. If I am not mistaken, keeping the `const ec_attribute_t` should make it so that when you do the `*some_attrib` later on you won't have to do a cast so we should be good if you remove this and switch back from the `const auto` to the `auto`?
```suggestion
std::optional<const ec_attribute_t> ec_util::get_attrib(uint8_t *buff, size_t len, ec_attrib_id_t id)
```